### DISCUSSION TITLE
Cannot logging!

### DISCUSSION THREAD

**Discussion by JavaZeroo** [nextcloud/all-in-one#961] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. My install code: 
`sudo docker run -it \
--name nextcloud-aio-mastercontainer \
--restart always \
-p 8080:8080 \
-e APACHE_PORT=11000 \
-e SKIP_DOMAIN_VALIDATION=true \
-e NEXTCLOUD_DATADIR="/media/jimmy/nextcloud" \
-e NEXTCLOUD_MOUNT="/media/" \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest`
2. After creating a new instance, I try to login in my nextcloud. But it stuck in the logging page,

### Expected behavior 
Login nextcloud. 
### Actual behavior 
spinning in circles :(
[**Screen Shot**](https://i.ibb.co/Y8Lc5mT/1a2af5a684d51643ac244490c98da8f.png)

### Host OS 
Ubuntu 22.04

#### Nextcloud AIO version 
1.5.0
#### Current channel 
leatest
#### Other valuable info 
I found that nextcloud container recive the 303 request, but I dont know whats wrong. SOSAAAAAAAAD

```
nc: getaddrinfo: Try again
Waiting for database to start...
              now              
-------------------------------
 2022-07-20 16:39:04.718241+02
(1 row)

Configuring Redis as session handler
Applying one-click-instance settings...
System config value one-click-instance set to boolean true
System config value one-click-instance.user-limit set to integer 100
Adjusting log files...
System config value logfile set to string /var/www/html/data/nextcloud.log
Config value logfile for app admin_audit set to /var/www/html/data/audit.log
Applying network settings...
System config value trusted_domains => 1 set to string www.jimmyisme.top:8080
System config value overwrite.cli.url set to string https://www.jimmyisme.top:8080/
System config value htaccess.RewriteBase set to string /
.htaccess has been updated
System config value files_external_allow_create_new_local set to boolean true
System config value trusted_proxies => 0 set to string 127.0.0.1
System config value trusted_proxies => 1 set to string ::1
Config value base_endpoint for app notify_push set to https://www.jimmyisme.top:8080/push
Error: Could not download app onlyoffice
System config value onlyoffice => jwt_secret set to string 4f0e787bd1549de3848380e7ac70ef9a06e78204a04bd03c
System config value onlyoffice => jwt_header set to string AuthorizationJwt
Config value DocumentServerUrl for app onlyoffice set to https://www.jimmyisme.top:8080/onlyoffice
System config value allow_local_remote_servers set to boolean true
[2022-07-20 22:40:16.122612 +08:00] WARN [rustls::conn] /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/rustls-0.20.2/src/conn.rs:1197: Sending fatal alert BadCertificate
[2022-07-20 22:40:16.123667 +08:00] ERROR [notify_push] src/main.rs:77: Self test failed: Error while communicating with nextcloud instance
[20-Jul-2022 22:40:16] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[20-Jul-2022 22:40:16] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[20-Jul-2022 22:40:16] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[20-Jul-2022 22:40:16] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[20-Jul-2022 22:40:16] NOTICE: fpm is running, pid 180
[20-Jul-2022 22:40:16] NOTICE: ready to handle connections
172.19.0.7 -  20/Jul/2022:22:40:30 +0800 "GET /index.php" 302
172.19.0.7 -  20/Jul/2022:22:40:50 +0800 "GET /index.php" 302
172.19.0.7 -  20/Jul/2022:22:40:58 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:40:58 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:40:58 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:40:58 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:40:58 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:40:58 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:40:59 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:41:01 +0800 "POST /index.php" 303 //This is my login try
172.19.0.7 -  20/Jul/2022:22:43:23 +0800 "GET /index.php" 302
172.19.0.7 -  20/Jul/2022:22:43:33 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:43:34 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:43:34 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:43:34 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:43:34 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:43:34 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:43:37 +0800 "POST /index.php" 303 //This is my login try
172.19.0.7 -  20/Jul/2022:22:44:57 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:44:57 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:44:57 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:44:57 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:44:58 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:44:58 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:45:00 +0800 "POST /index.php" 303 //This is my login try
172.19.0.7 -  20/Jul/2022:22:47:50 +0800 "GET /index.php" 200
172.19.0.7 -  20/Jul/2022:22:47:56 +0800 "POST /index.php" 303 //This is my login try
```

**Comment by Zoey2936** [comment]
Whats your reverse proxy setup?

**Comment by Zoey2936** [comment]
(And docker is NOT a host os)

**Comment by dantefromhell** [comment]
The OPs log looks like something similar I have been observing:
User logins in, no error message is shown but the users ends up back at the login screen.

Looking at the network traffic through the browser utilities I can see the following happen
- HTTP Post to `/login` answered with HTTP/303 to `/dashboard`
- HTTP GET to `/dashboard` answered with HTTP/303 to `/login?redirect_url=/apps/dashboard/`

Clearing cookies, browser local storage, ... does not resolve the issue. 
I can reproduce in different Browsers (Brave, Firefox, Edge [Chrome based]).

The only working solution so far is to try again & again. Sometime > 10 times until successful login.

**Comment by Zoey2936** [comment]
And why are you opening the cloud over an ip using a port, when in your logs a domain with an other port is written? And also your are using http in your browser, while in the logs https is written? I am a bit confused...

**Comment by JavaZeroo** [comment]
> Whats your reverse proxy setup?

Same as the doc:
```
:80 {
        # Set this path to your site's directory.
        root * /usr/share/caddy

        # Enable the static file server.
        file_server

        # Another common task is to set up a reverse proxy:
        # reverse_proxy localhost:8080

        # Or serve a PHP site through php-fpm:
        # php_fastcgi localhost:9000
}
:88 {
    reverse_proxy localhos:11000
}
https://my.domain:443 {
    reverse_proxy localhost:11000
}
https://my.domain:88 {
    reverse_proxy localhost:11000
}
```

**Comment by JavaZeroo** [comment]
> And why are you opening the cloud over an ip using a port, when in your logs a domain with an other port is written? And also your are using http in your browser, while in the logs https is written? I am a bit confused...

Thats a long story. Since chinese ISP **block 443 and 80 port for home user**. China's network environment is too different, and I have tried to solve many problems myself. At the same time, since I use an external hard drive to store the device, I can't use `snap install nextcloud` to install nextcloud (Not enough customization options), so nextcloud-aio is the easiest way I found to install nextcloud. I fix the problem with another issues few days ago. You can check this out #943 

btw thank you so much for helping me find out the issues!!!!! =3=

**Comment by JavaZeroo** [comment]
> The OPs log looks like something similar I have been observing: User logins in, no error message is shown but the users ends up back at the login screen.
> 
> Looking at the network traffic through the browser utilities I can see the following happen
> 
> * HTTP Post to `/login` answered with HTTP/303 to `/dashboard`
> * HTTP GET to `/dashboard` answered with HTTP/303 to `/login?redirect_url=/apps/dashboard/`
> 
> Clearing cookies, browser local storage, ... does not resolve the issue. I can reproduce in different Browsers (Brave, Firefox, Edge [Chrome based]).
> 
> The only working solution so far is to try again & again. Sometime > 10 times until successful login.

Unfortunately, I tried thousands of times without success :(

**Comment by szaimen** [comment]
Moving to discussions as I cannot reproduce this

**Comment by szaimen** [comment]
Sorry but I cannot reproduce this. Is something useful in the Nextcloud logs maybe?

**Comment by szaimen** [comment, marked as answer]
It seems like you did not set it up correctly. Did you follow https://github.com/nextcloud/all-in-one#how-to-get-it-running-using-the-acme-dns-challenge correctly? If yes, you should access your Nextcloud using your domain and not your ip-address.

**Reply by dantefromhell** [reply]
My data-point:
Yes I have followed instructions correctly and not been using ACME DNS challenge nor IPs.

Also no information in the nextcloud logs...
